xbmc/games/addons/input/GameClientInput.cpp

```diff
@@ -82,15 +83,14 @@ void CGameClientInput::Start(IGameInputCallback* input)
     OpenMouse(it->ActiveController().Controller());
   }
 
-  // Open joysticks
-  //! @todo Move to player manager
-  for (const auto& port : controllers.Ports())
+  // Connect/disconnect active controllers
+  for (const CPortNode& port : controllers.Ports())
   {
-    if (port.PortType() == PORT_TYPE::CONTROLLER && !port.CompatibleControllers().empty())
-    {
-      ControllerPtr controller = port.ActiveController().Controller();
-      OpenJoystick(port.Address(), controller);
-    }
+    const ControllerPtr activeController = port.ActiveController().Controller();
```

Comment:
You only need `activeController` if `port.Connected()` is `true` so maybe move it into the inner scope?

Reply:
Fixed